src/game_api/state_structs.hpp

```diff
@@ -4,6 +4,7 @@
 #include <cstdint>
 
 #define MAX_PLAYERS 4
+using ENT_TYPE = uint32_t; // NoAlias
```

Comment:
This is already defined in entity.hpp, can that not be included here?

Reply:
it's not in entity.hpp, it's in lua_backend.hpp, with all the others, it's in here cuz that's covers all the places that don't reference lua_backend.hpp, there is no way to cross include between, exceptionally we could make new header file for all types used across the API, maybe throw some #define there as well and include that everywhere

Reply:
It is here https://github.com/spelunky-fyi/overlunky/pull/136/files#diff-4cebcfa05af7753384df6a6b719e6007fa7dd5336bf20937fc1cd86c29911bafR52
Indeed with the one you mentioned there are 3 places this type alias lives. I think it would be better to add them to a separate file, but it's not a bug to have multiple so no hurry to fix that, we can do it some other time so we don't delay merging this more.